src/main/java/hudson/plugins/git/GitSCM.java

```diff
@@ -1106,7 +1143,7 @@ public void checkout(Run<?, ?> build, Launcher launcher, FilePath workspace, Tas
             ext.beforeCheckout(this, build, git, listener);
         }
 
-        retrieveChanges(build, git, listener);
+        retrieveChanges(build, git, listener, GitUtils.workspaceToNode(workspace));
```

Comment:
If `Jenkins.getInstance()` returns null it means that you are making the call from inside an agent JVM, signalling a fundamental design mistake somewhere.